src/VisualStudio/Core/Def/Implementation/ProjectSystem/RuleSets/VisualStudioRuleSetManager.RuleSetFile.cs

```diff
@@ -24,47 +29,53 @@ private sealed class RuleSetFile : IRuleSetFile
 
             private Exception _exception;
 
-            public RuleSetFile(string filePath, IVsFileChangeEx fileChangeService, VisualStudioRuleSetManager ruleSetManager)
+            public RuleSetFile(string filePath, VisualStudioRuleSetManager ruleSetManager)
             {
-                _filePath = filePath;
+                FilePath = filePath;
                 _ruleSetManager = ruleSetManager;
+            }
 
-                ImmutableArray<string> includes;
-
-                try
-                {
-                    includes = RuleSet.GetEffectiveIncludesFromFile(filePath);
-                }
-                catch (Exception e)
+            internal void InitializeFileTracking(IVsFileChangeEx fileChangeService)
+            {
+                lock (_gate)
```

Comment:
I believe fileChangeService is now truely free-thread?

Reply:
It is, but we could have multiple callers of this at the same time if we have multiple projects trying to use the same file.